src/settings/LoanHistory/utils/normalize.js

```diff
@@ -0,0 +1,48 @@
+import {
+  cloneDeep,
+  forEach,
+  set,
+  isArray,
+  isString,
+} from 'lodash';
+import { change } from 'redux-form';
+
+import { closingTypesMap } from '../../../constants';
+
+const checkInterval = ({ data, dispatch }) => {
+  const newData = cloneDeep(data);
+
+  forEach(Object.entries(data.closingType), ([key, value]) => {
+    if (isString(value) && value !== closingTypesMap.INTERVAL) {
+      set(newData, key, {});
+      dispatch(change('loanHistoryForm', key, {}));
+    } else if (isArray(value)) {
+      forEach(value, (loanExceptionsValue, index) => {
+        if (loanExceptionsValue !== closingTypesMap.INTERVAL) {
+          set(newData, [key, index], {});
+          dispatch(change('loanHistoryForm', [key, index], {}));
+        }
+      });
+    }
+  });
+  return newData;
+};
+
+const filter = (entity, ...callbacks) => {
+  let filteredEntity = cloneDeep(entity);
+
+  forEach(callbacks, callback => {
+    filteredEntity = callback(filteredEntity);
+  });
+
+  return filteredEntity;
+};
+
+
+export const normalize = entity => {
+  const callbacks = [checkInterval];
```

Comment:
I'm not sure, that callbacks is a right word to describe the array purposes. Maybe, it should be renamed to "handlers", for example.